src/pip/_internal/commands/show.py

```diff
@@ -140,41 +144,86 @@ def get_requiring_packages(package_name):
         yield package
 
 
-def print_results(distributions, list_files=False, verbose=False):
+def get_package_info(distributions, list_files=False, verbose=False):
     """
-    Print the information from installed distributions found.
+    Build a dictionary with  information from installed distributions found.
     """
-    results_printed = False
+    pkg_infos_list = []
+
     for i, dist in enumerate(distributions):
-        results_printed = True
-        if i > 0:
-            write_output("---")
 
-        write_output("Name: %s", dist.get('name', ''))
-        write_output("Version: %s", dist.get('version', ''))
-        write_output("Summary: %s", dist.get('summary', ''))
-        write_output("Home-page: %s", dist.get('home-page', ''))
-        write_output("Author: %s", dist.get('author', ''))
-        write_output("Author-email: %s", dist.get('author-email', ''))
-        write_output("License: %s", dist.get('license', ''))
-        write_output("Location: %s", dist.get('location', ''))
-        write_output("Requires: %s", ', '.join(dist.get('requires', [])))
-        write_output("Required-by: %s", ', '.join(dist.get('required_by', [])))
+        pkg_info = {}
+
+        pkg_info["Name"] = dist.get('name', '')
+        pkg_info["Version"] = dist.get('version', '')
+        pkg_info["Summary"] = dist.get('summary', '')
+        pkg_info["Home-page"] = dist.get('home-page', '')
+        pkg_info["Author"] = dist.get('author', '')
+        pkg_info["Author-email"] = dist.get('author-email', '')
+        pkg_info["License"] = dist.get('license', '')
+        pkg_info["Location"] = dist.get('location', '')
+        pkg_info["Requires"] = dist.get('requires', [])
```

Comment:
Is this Requires field a list of PEP 508 specifiers? Should this be renamed `Requires-Dist` so the json field names wil align with the core metadata spec?

Reply:
Hi @sbidoul 

I am not aware of the fact if `Requires` field is in the list of PEP 508 specifiers. I did find that accordin to the [core metadata spec](https://packaging.python.org/specifications/core-metadata/), the usage of `Requires` is similar to `Requires-Dist` and the usage of `Required-by` is similar to `Requires-External`.

Maybe @xavfernandez can help shed some light on it, and I can update the names to match the core metadata spec in another PR?

Reply:
> Should this be renamed Requires-Dist so the json field names wil align with the core metadata spec?
It could but this would be a breaking change, we might want to expose the same information in both `Requires` & `Requires-Dist` for a few pip version. It would definitely happen in an other PR.

Reply:
>  It could but this would be a breaking change, we might want to expose the same information in both `Requires` & `Requires-Dist` for a few pip version. It would definitely happen in an other PR.

Sure than I will keep the keys as-in and won't touch them.